lib/edamam_api_wrapper.rb

```diff
@@ -0,0 +1,42 @@
+require 'httparty'
+
+class EdamamApiWrapper
+  URL = "https://api.edamam.com/search"
+  URL2 = "http://www.edamam.com/ontologies/edamam.owl#recipe_"
+  APP_ID = ENV["APP_ID"]
+  APP_KEY = ENV["APP_KEY"]
+
+
+
+  def self.list_recipes(ingredient)
+    response = HTTParty.get("#{URL}?q=#{ingredient}&app_id=#{APP_ID}&app_key=#{APP_KEY}&from=0&to=100")
+
+    recipes = []
+
+    if response["hits"]
+      response["hits"].each do |recipe|
+        recipes << Recipe.new(recipe["recipe"]["label"], recipe["recipe"]["uri"], recipe["recipe"]["ingredientLines"], recipe["recipe"]["image"], recipe["recipe"]["healthLabels"],
+          recipe["recipe"]["url"],
+          recipe["recipe"]["source"])
+        end
+      end
+      return recipes
+    end
+
+    def self.get_recipe(id)
+      encode = URI.encode("#{URL}?r=#{URL2}#{id}&app_id=#{APP_ID}&app_key=#{APP_KEY}")
+
+      response = HTTParty.get(encode)
+
+      if response.empty?
+        
+        return nil
+
+      else
+
```

Comment:
You need to verify that the response has the keys below, for invalid ids this will crash the app.